### PR TITLE
Add OpenNavSurf BAG recipe

### PR DESCRIPTION
**Pull request by selimnairb**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

@conda-forge-admin, please ping conda-forge/help-c-cpp, ready for review!
@conda-forge-admin, please ping conda-forge/help-python, ready for review!
@conda-forge-admin, please ping conda-forge/help-python-c, ready for review!

### CODE REVIEW

**Comment by selimnairb** [comment]
macOS build, which once was working, is now failing with this error message while installing build packages in mamba:
```
...
KeyError: 'https://conda.anaconda.org/conda-forge'
...
```

Not sure there is anything I can do about this. Is it a transient error that just sometimes happens?

**Comment by selimnairb** [comment]
Builds working again after transient build error last week. @conda-forge-admin, please ping conda-forge/help-c-cpp

**Comment by selimnairb** [comment]
Hello, @conda-forge-admin, please ping conda-forge/help-c-cpp to ask for review. Thanks!

File: .ci_support/osx_arm64.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+c_compiler:
```
Comment (xhochy) [inline .ci_support/osx_arm64.yaml:1]: Please remove this file.

File: recipes/opennavsurf-bag/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    - hdf5
+    - libxml2
+    - python
+    - pip
+    - setuptools
+    - 'setuptools-scm[toml]'
+    - wheel
+    - cmake-build-extension
+  run:
+    - hdf5
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:44]: Should come in via run_exports 
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    - libxml2
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:45]:
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    - 'setuptools-scm[toml]'
+    - wheel
+    - cmake-build-extension
+  host:
+    - hdf5
+    - libxml2
+    - python
+    - pip
+    - setuptools
+    - 'setuptools-scm[toml]'
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:40]: The [] syntax has a different meaning in conda.
```suggestion
    - setuptools-scm
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+requirements:
+  build:
+    - cmake
+    - ninja
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - sysroot_linux-64  # [linux64]
+    - python  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - pip
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:26]: Pleasure remove all the (pure) Python dependencies from build. They should be in host.

File: recipes/opennavsurf-bag/build.sh [hunk 6]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+set -ex # Abort on error.
+
+mkdir build
+
+# Configure CMake build
+cmake -G Ninja \
```
Comment (xhochy) [inline recipes/opennavsurf-bag/build.sh:8]:
```suggestion
cmake ${CMAKE_ARGS} -G Ninja \
```

File: recipes/opennavsurf-bag/build.sh [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+  -DCMAKE_BUILD_TYPE=Release \
+  -B build -S . \
+  -DCMAKE_PREFIX_PATH=$PREFIX \
```
Comment (xhochy) [inline recipes/opennavsurf-bag/build.sh:11]:
```suggestion
```

File: recipes/opennavsurf-bag/build.sh [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+  -DCMAKE_INSTALL_PREFIX=$PREFIX \
```
Comment (xhochy) [inline recipes/opennavsurf-bag/build.sh:12]:
```suggestion
```

File: recipes/opennavsurf-bag/build.sh [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  -DCMAKE_INSTALL_LIBDIR=lib \
```
Comment (xhochy) [inline recipes/opennavsurf-bag/build.sh:13]:
```suggestion
```

File: recipes/opennavsurf-bag/bld.bat [hunk 10]
```diff
@@ -0,0 +1,41 @@
+setlocal EnableDelayedExpansion
+
+mkdir build
+if errorlevel 1 exit 1
+
+if  %vc% LEQ 9 set MSVC_VER=1500
```
Comment (xhochy) [inline recipes/opennavsurf-bag/bld.bat:6]: I don't think these settings are needed. We have not been building for old VCs for a long time.

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by selimnairb** [comment]
@xhochy Thank so much for the review! I made the requested changes. I had to add `libcxx  # [osx]` to the `run` requirements as the macOS builds were failing without. Please let me know if there are any other changes needed. Thanks!

File: .ci_support/osx_arm64.yaml (hunk 1 above)
Comment (selimnairb) [reply .ci_support/osx_arm64.yaml]: Resolved.

File: recipes/opennavsurf-bag/bld.bat (hunk 10 above)
Comment (selimnairb) [reply recipes/opennavsurf-bag/bld.bat]: Resolved.

**Review by carterbox — CHANGES_REQUESTED** [review]
This package is shipping C header files and a non-python shared lib. It looks like the Python parts and C parts are build separtely, so could you please add an additional package `libbaglib` to this PR and put the C library and headers there. Then have the python package depend on this new package? This way the C shared lib will only be built once (instead of with each python version) and C users will not need to install the python interpreter.

Also, the new libbaglib package will need to run_export itself with appropriate ABI compatability pinning.

File: recipes/opennavsurf-bag/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    - libcxx  # [osx]
+    - python
+
+test:
+  files:
+    - run_test.py
+  imports:
+    - bagPy
+  commands:
+    - python run_test.py
```
[the remark this answers is not in the document]
Comment (selimnairb) [reply recipes/opennavsurf-bag/meta.yaml:50]: Sure, I will add a second package. Is there an exemplary meta.yaml that you would recommend I reference (and/or documentation). I am somewhat familiar with [GDAL's](https://github.com/conda-forge/gdal-feedstock/blob/main/recipe/meta.yaml), is that a good example?

[the remark this answers is not in the document]
Comment (selimnairb) [reply recipes/opennavsurf-bag/meta.yaml]: @carterbox I split out the C++ library into subpackage libbaglib and the Python bindings into subpackage bagpy. I was able to build locally using conda-build, however the CI builds for all three platforms are failing with the following error:

```
['  ERROR (bagpy,lib/python3.11/site-packages/bagPy/_bagPy.so): $RPATH/libbaglib.so not found in packages, sysroot(s) nor the missing_dso_whitelist.\n.. is this binary repackaging?']
```
 From [line 842](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=811351&view=logs&jobId=6f142865-96c3-535c-b7ea-873d86b887bd&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823) of the logs.

Do you know how to fix this? Is this something wrong in meta.yaml, or the underlying CMake config?

Thanks.

[the remark this answers is not in the document]
Comment (selimnairb) [reply recipes/opennavsurf-bag/meta.yaml]: @carterbox Nevermind. I figured out the error: needed to have the libbaglib sub-package listed as a host requirement for the Python bindings (in addition to being a run requirement). I missed that in the first attempt. On to fixing the next error...

[the remark this answers is not in the document]
Comment (selimnairb) [reply recipes/opennavsurf-bag/meta.yaml]: @carterbox I have resolved build issues. Resolving this conversation. Please let me know if there are other revisions needed. Thanks!

File: recipes/opennavsurf-bag/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,109 @@
...
+    - {{ compiler('cxx') }}
+    - sysroot_linux-64  # [linux64]
+    - python  # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - swig
+  host:
+    - hdf5
+    - libxml2
+  run:
+    - libcxx  # [osx]
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:31]: This is no longer needed (was a bug in the compiler package)
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 13]
```diff
@@ -0,0 +1,109 @@
...
+  run_exports:
+    - hdf5
+
+requirements:
+  build:
+    - cmake
+    - ninja
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - sysroot_linux-64  # [linux64]
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:23]: Only add this if you require a specific version.
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 14]
```diff
@@ -0,0 +1,109 @@
...
+
+source:
+  url: https://github.com/OpenNavigationSurface/BAG/archive/refs/tags/release-{{ version }}.tar.gz
+  sha256: 7b246eec04f46782b74bc9bf14be5ddf0a157360d17bd2f48a150992c540025d
+
+build:
+  number: 0
+  skip: true  # [py<38]
+  run_exports:
+    - hdf5
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:15]:
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 15]
```diff
@@ -0,0 +1,109 @@
...
+    build:
+      run_exports:
+        - {{ pin_subpackage('libbaglib', max_pin='x.x') }}
+    requirements:
+      build:
+        - cmake
+        - ninja
+        - {{ compiler('c') }}
+        - {{ compiler('cxx') }}
+        - sysroot_linux-64  # [linux64]
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:46]:
```suggestion
```

File: recipes/opennavsurf-bag/meta.yaml [hunk 16]
```diff
@@ -0,0 +1,109 @@
...
+        - catch2
+      host:
+        - hdf5
+        - libxml2
+      run:
+        - libcxx  # [osx]
```
Comment (xhochy) [inline recipes/opennavsurf-bag/meta.yaml:52]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Review by xhochy — APPROVED** [review]

**Pull request merged**